netket/optimizer/qgt/qgt_onthefly.py

```diff
@@ -72,26 +83,13 @@ class QGTOnTheFlyT(LinearOperator):
     the field `sr`.
     """
 
-    apply_fun: Callable[[PyTree, jnp.ndarray], jnp.ndarray] = struct.field(
-        pytree_node=False, default=Uninitialized
-    )
-    """The forward pass of the Ansatz."""
+    mat_vec: Callable[[PyTree, float], PyTree] = Uninitialized
+    """The S matrix-vector product as generated by mat_vec_factory.
+    It's a jax.Partial, so can be used as pytree_node."""
```

Comment:
Now that i think of it, all those fields would better be inernal fields... (_mat_vec, _params)

Reply:
because they are not part of the public api (for example, you just removed several)/